internal/cephfs/fsjournal.go

```diff
@@ -69,18 +82,28 @@ func checkVolExists(ctx context.Context, volOptions *volumeOptions, secret map[s
 	}
 	imageUUID := imageData.ImageUUID
 	vid.FsSubvolName = imageData.ImageAttributes.ImageName
+	var evnf ErrVolumeNotFound
+
+	clone, cloneInfoErr := getcloneInfo(ctx, volOptions, cr, volumeID(vid.FsSubvolName))
+	if cloneInfoErr != nil {
+		if !errors.As(cloneInfoErr, &evnf) {
+			return nil, status.Error(codes.Internal, cloneInfoErr.Error())
+		}
+	} else {
+		if clone.Status.State != cephFSCloneCompleted {
```

Comment:
Clone can fail and hence never reach completed status. We should check clone status here to ensure it is has not failed, and if it failed handle the removal of the clone (rm --force) and the other artifacts as required.